distributed/semaphore.py

```diff
@@ -460,6 +462,26 @@ def acquire(self, timeout=None):
         timeout = parse_timedelta(timeout)
         return self.client.sync(self._acquire, timeout=timeout)
 
+    async def _release(self):
+        # popleft to release the oldest lease first
+        lease_id = self._leases.popleft()
+        logger.info("%s releases %s for %s", self.client.id, lease_id, self.name)
+
+        try:
+            return await retry_operation(
+                self.client.scheduler.semaphore_release,
+                name=self.name,
+                lease_id=lease_id,
+                operation="semaphore release: client=%s, lease_id=%s, name=%s"
+                % (self.client.id, lease_id, self.name),
+            )
+        except OSError:  # Too many broken connections, release fails
+            logger.error(
+                "Release failed for client=%s, lease_id=%s, name=%s. Cluster network might be unstable."
```

Comment:
We should log the original exception. Otherwise this is impossible to debug if things don't go smooth. Suspecting the cluster network is only one possibility and I wouldn't jump to any conclusions but just log what happened.